Sources/MessagesDisplayDataSource.swift

```diff
@@ -25,41 +25,52 @@
 import Foundation
 
 public protocol MessagesDisplayDataSource: class, MessagesDataSource {
-
-    func messageColor(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor
-
-    func avatar(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> Avatar
-
+    
+    func textColor(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor
+    
+    func backgroundColor(for message: MessageType, at  indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor
+    
+    func avatar(for message: MessageType, at  indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> Avatar
+    
     func messageHeaderView(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> MessageHeaderView?
-
+    
     func messageFooterView(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> MessageFooterView?
-
+    
     func cellTopLabelAttributedText(for message: MessageType, at indexPath: IndexPath) -> NSAttributedString?
-
+    
     func cellBottomLabelAttributedText(for message: MessageType, at indexPath: IndexPath) -> NSAttributedString?
-
+    
 }
 
-public extension MessagesDisplayDataSource {
-
-    func messageColor(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor {
+// "where Self: MessagesViewController" make these methods a `default` implimentation for any view that subclasses of MessagesViewController
+public extension MessagesDisplayDataSource where Self: MessagesViewController {
```

Comment:
I think this was better without the `Self: MessagesViewController` requirement, because now you only get default behavior for `MessagesViewController`. 

If the `MessagesDisplayDataSource` is another object, you get no defaults. The original implementation gives you defaults for any object because the `MessagesDisplayDataSource` is not dependent on `MessagesCollectionView` -- only the `MessagesDataSource`. 

It's very possible to use this library without the `MessagesViewController` and just the `MessagesCollectionView` in your own controller. It's also very possible to use a view model helper object as the `MessagesDisplayDataSource`.